xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -1854,10 +1854,19 @@ CacheInfo CVideoPlayer::GetCachingTimes()
 
 void CVideoPlayer::HandlePlaySpeed()
 {
+  auto cacheIdealSize = CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_cacheMemSize;
+  bool cacheToMemory = !!cacheIdealSize;
```

Comment:
1. `const bool` to make clear to the reader that you will not change the value anymore after first assignment.
2. I know very well the meaning of `!!`to convert a numeric value to a bool, but don't consider it good style. Looks cute, but many people out there do not understand this and I have even seen people treating this a typo suggesting to remove one of the exclamation marks. This "confusion" can easily avoided by not using such "tricks".